### ISSUE TITLE
Disconnection/Socket issue

### ISSUE THREAD

**Issue by DocileDanny** [Stary2001/InputRedirection#8]
In NTR mode, there are situations where InputRedirection will die, while NTRViewer, and other NTR-based applications continue to work. e.g. Nitro Stream + Mem patch will allow Nitro Stream to be used in Pokémon X/Y with wifi in-game disabled. However, doing so causes a "connection closed" message to appear, causing InputRedirection to stop working. Additionally, wifi drops/disconnections while in other games will cause a similar issue, NTRViewer will be able to reconnect and continue operating, but InputRedirection will no longer function, requiring a 3DS reboot.

Not knowing C++, I can't fix the issue myself, but I think the problem is here:

https://github.com/Stary2001/InputRedirection/blob/master/input_proc/source/main.c

on line 92, if(byte_count == -1) you break, causing the loop to exit, after which InputRedirection closes completely. What should happen instead is either some type of continue, where you ignore this state, or an attempt to rebind the socket. I'm not sure on the socket type being used, so it's contingent on that, as to how you'd fix it, but basically you'll want to destroy the old socket and attempt to rebind or recreate the socket and continue the loop as normal. You basically don't want the inputloop to ever end.

TL;DR
The important information is that on a socket failure there's no attempt to re-create or re-establish the connection. Doesn't seem like it would be *too* hard to fix, I'd attempt it myself I'm not sure how to compile this for the 3DS, nor do I know c++, as mentioned earlier. :)

Other then that small thing this works really well! :)

[edit]
I don't know C++, as previously mentioned, so I can't test this, but based on some quick research/socket knowledge from other languages,, this is the type of thing input_loop should be doing, but it needs to be tested, and probably fixed, but hopefully you can see what I was trying to make it do.

http://pastebin.com/Lmu4mC9P

**Comment by Stary2001** [comment]
NTR mode doesn't use that code :)
Good suggestion for HBL mode, though, I'll look into it when I have time.

**Comment by DocileDanny** [comment]
Whoops, I got confused by ifndef ; I thought it was a positive condition, I see now that it's a negative. In that case, the cause of the issue this is a ticket for is a total mystery to me, but I'm glad I gave you at least one useful suggestion. :)

**Comment by Stary2001** [comment]
HBL code is now gone... so fixed? sorta?
